foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_8_0/AiopsReferencePDPPythonMigration.java

```diff
@@ -0,0 +1,122 @@
+
+/**
+ Baton migration class to verify whether a migration to the Policy Decision Point (Python) module files is required from
+ the aiops reference refactor.
+ **/
+public class AiopsReferencePDPPythonMigration extends AbstractAissembleMigration {
+
+    public static final Logger logger = LoggerFactory.getLogger(AiopsReferencePDPPythonMigration.class);
+    public static final Map<String, String> AIOPS_REFERENCE_PYTHON_PROPERTIES_FILE_PACKAGE_MAP = Map.of(
+            "policy-decision-point.aiops-security.properties.vm", "policy-decision-point.aissemble-security.properties.vm",
+            "aiops-security.properties", "aissemble-security.properties",
+            "aiops.authority","aissemble.authority",
+            "aiopskey","aissemblekey",
+            "/deployments/aiops-secure.jks","/deployments/aissemble-secure.jks"
+            );
+
+    /**
+     * Function to check whether the migration is necessary.
+     * @param file file to check
+     * @return shouldExecute - whether the migration is necessary.
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = false;
+
+        try {
+            if(file.getName().equals("aiops-security.properties")) {
+                shouldExecute = true;
+            } else {
+                BufferedReader aiopsReferencePythonDockerfileConfig = new BufferedReader((new FileReader(file)));
+
+                String line;
+                while((line = aiopsReferencePythonDockerfileConfig.readLine()) !=null && !shouldExecute) {
+                    for (String key : AIOPS_REFERENCE_PYTHON_PROPERTIES_FILE_PACKAGE_MAP.keySet()) {
+                        if (line.contains(key)) {
+                            shouldExecute = true;
+                        }
+                    }
+                }
+            }
+        } catch (IOException e) {
+            logger.error("Error in determining whether a 'AIOPS' renamed properties file requires a migration.");
+        }
+
+        return shouldExecute;
+    }
+
+    /**
+     * Performs the migration if the shouldExecuteOnFile() returns true.
+     * @param file file that will be getting refactored.
+     * @param refactoredName name in which the file will be refactored to.
+     * @return isRefactoredSuccessfully - Whether the file has been refactored successfully.
+     */
+    private static boolean fileNameRefactor(File file, String refactoredName) {
+        boolean isRefactoredSuccessfully = false;
+        try {
+            // Construct new file path
+            File renamedFile = new File(file.getParent(), refactoredName);
+
+            // Rename the file
+            isRefactoredSuccessfully = file.renameTo(renamedFile);
+        } catch (Exception e) {
+            logger.error("Error in refactoring aiops-security.properties.");
+        }
+
+        return isRefactoredSuccessfully;
+    }
+
+    /**
+     * Performs the migration if the shouldExecuteOnFile() returns true.
```

Comment:
same comment as above.  This method can be called regardless of what shouldExecuteOnFile() returns

Reply:
Updated the description to better describe functionality.